[DOCS] Adds Connecting section to docs (#6183)
elastic/elasticsearch-net: docs/connecting.asciidoc

lifetime. Internally the client manages and maintains HTTP connections to nodes, 
reusing them to optimize performance. If you use a dependency injection 
container for your application, the client instance should be registered with a 
singleton lifetime.

[discrete]
[[cloud-deployment]]
=== Connecting to a cloud deployment

Connecting to an Elasticsearch Service deployment is achieved by providing the 
unique Cloud ID for your deployment when configuring the `ElasticsearchClient` 
instance. You can retrieve the Cloud ID from the homepage of the deployment in 
Elasticsearch Service. You also require suitable credentials that your 
application uses to authenticate with your deployment.

As a security best practice, it is recommended to create a dedicated API key per 
application, with permissions limited to only those required for any API calls 
the application is authorized to make.

The following snippet shows you how to create a client instance that connects to 
an {es} deployment in the cloud.

[source,csharp]
----
using Elastic.Clients.Elasticsearch;
using Elastic.Transport;

var client = new ElasticsearchClient("<CLOUD_ID>", new ApiKey("<API_KEY>"));
----

> Replace the placeholder string values above with your cloud ID and the API key 

Reviewer: I think we should make this one of those numbered code comments (not sure what we call them), with the blue numbered circle as this directly references the line of code it relates to?

configured for your application to access your deployment.

[discrete]
[[single-node]]
=== Connecting to a single node

Single node configuration is best suited to connections to a multi-node cluster 
running behind a load balancer or reverse proxy, which is exposed via a single 
URL. It may also be convenient to use a single node during local application 
development. If the URL represents a single {es} node, be aware that this offers 
no resiliency should the server be unreachable or unresponsive.

> By default, with security features like authentication and TLS enabled. When you 
> start {es} for the first time, TLS is configured automatically for the HTTP 
> layer. A CA certificate is generated and stored on disk that is used to sign the 
> certificates for the HTTP layer of the {es} cluster.

Reviewer: ```suggestion
By default, security features such as authentication and TLS are enabled on {es} clusters. 
When you start {es} for the first time, TLS is configured automatically for the HTTP 
layer. A CA certificate is generated and stored on disk which is used to sign the 
certificates for the HTTP layer of the {es} cluster.
```

In order for the client to establish a connection with the cluster over HTTPS, 
the CA certificate must be trusted by the client application. There are several 

> mechanisms for working with certificates, but the simplest choice is to use the 

Reviewer: We need a link here:

```suggestion
mechanisms for <<working-with-certificates, working with certificates>>, but the simplest choice is to use the 
```
